oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCheckCommand.java

```diff
@@ -526,7 +551,19 @@ public void traverse() throws IOException {
 
             try {
                 writer = Files.newWriter(references, Charsets.UTF_8);
-                traverseChildren(nodeStore.getRoot(), "/", writer, count);
+                if (paths.length == 0) {
+                    traverseChildren(nodeStore.getRoot(), PATH_DELIM, writer, count);
+                } else {
+                    for (String path: paths ) {
+                        String[] nodeList = path.split(PATH_DELIM);
+                        NodeState state = nodeStore.getRoot();
+                        for (String node: nodeList) {
+                            if (node.isEmpty()) continue;
```

Comment:
I don't think that's needed... That would mean the path is something like "a//b", right? Instead of ignoring the empty entry, I would throw an exception, or do what PathUtils does normally.

Reply:
Actually "/oak:index".split("/") would return ["","oak:index"] -> so this ignores the first empty entry there.

Reply:
I understand. You are right!